src/BaGet.Core/Search/DatabaseSearchService.cs

```diff
@@ -106,15 +106,32 @@ public DatabaseSearchService(IContext context, IFrameworkCompatibilityService fr
             };
         }
 
-        public Task<AutocompleteResponse> ListPackageVersionsAssync(
-            string packageId,
-            bool includePrerelease,
-            bool includeSemVer2,
+        public async Task<AutocompleteResponse> ListPackageVersionsAsync(
+            VersionsRequest request,
             CancellationToken cancellationToken)
         {
-            // TODO: Support versions autocomplete.
-            // See: https://github.com/loic-sharma/BaGet/issues/291
-            throw new NotImplementedException();
+            var packageId = request.PackageId.ToLower();
+            IQueryable<Package> search = _context
+                .Packages
+                .Where(p => p.Id.ToLower().Equals(packageId));
+
+            search = AddSearchFilters(
+                search,
+                request.IncludePrerelease,
+                request.IncludeSemVer2,
+                packageType: null,
+                frameworks: null);
+
+            var results = await search
+                .Select(p => p.Version.ToNormalizedString())
```

Comment:
Please avoid using the `p.Version` property as it actually parses the `OriginalVersionString` property. Also, I'm not sure this plays nicely with EF Core, it may require client side evaluation (as the `ToNormalizedString` method can't be translated to SQL). Please use the `p.NormalizedVersionString` property instead:

```suggestion
                .Select(p => p.NormalizedVersionString)
```